meteor/client/ui/Settings/ShowStyle/VariantSettings.tsx

```diff
@@ -30,22 +44,66 @@ interface IShowStyleVariantsProps {
 interface IShowStyleVariantsSettingsState {
 	editedMappings: ProtectedString<any>[]
 	timestampedFileKey: number
+	dndVariants: ShowStyleVariant[]
 }
 
+interface DragVariant {
+	index: number
+	type: ShowStyleDragDropTypes
+}
+
+const SET_STATE_DELAY = 50
+
 export const ShowStyleVariantsSettings = withTranslation()(
 	class ShowStyleVariantsSettings extends React.Component<
 		Translated<IShowStyleVariantsProps>,
 		IShowStyleVariantsSettingsState
 	> {
+		private timer?: number
+
 		constructor(props: Translated<IShowStyleVariantsProps>) {
 			super(props)
 
 			this.state = {
 				editedMappings: [],
 				timestampedFileKey: Date.now(),
+				dndVariants: this.props.showStyleVariants,
+			}
+		}
+		componentDidUpdate(prevProps: Readonly<Translated<IShowStyleVariantsProps>>) {
+			if (this.showStyleVariantsChanged(prevProps) || this.noShowStyleVariantsPresentInState()) {
+				this.timer = Meteor.setTimeout(() => {
+					this.setState({
+						dndVariants: this.props.showStyleVariants,
+					})
+				}, SET_STATE_DELAY)
+			} else if (this.timer) {
+				Meteor.clearTimeout(this.timer)
+			}
+		}
+
+		componentWillUnmount() {
+			if (this.timer) {
+				Meteor.clearTimeout(this.timer)
 			}
 		}
 
+		private showStyleVariantsChanged = (prevProps: Readonly<Translated<IShowStyleVariantsProps>>): boolean => {
```

Comment:
Is moving variants in a separate window (tab / machine etc.) not considered a change that should be reflected in other clients? In Sofie, we're generally used to data changes spreading automatically to all the clients, so this might be counterintuitive and even result in losing the order in some very rare cases

Reply:
I'm not sure i understand, if you have two tabs open with the variant table, they will both update in reaction to any drag and drop change instantly. Whenever the order is changed in the database, the table reorders in the UI.

Reply:
Maybe we are doing something different because i believe it worked for me before i left, i will dive into it tomorrow when i get to work! Thanks for showing me!

Reply:
I fixed this by checking if the previous showstylevariants match the current ones present in state. After testing on multiple tabs i could see the update happening instantly.

Reply:
Note that your new condition `prevShowStyleVariants !== this.props.showStyleVariants` will be true if the `trackerAutorun` function of the tracker wrapping the `ShowStyleBaseSettings` component reruns. Every time it runs, a new array of showStyleVariants is returned, even though the actual data might not have changed. I don't know if that's what you intended.